File: code/ch10/10.2.1-house-rob.java
class Solution {
    public int helper(int[] nums, int start) {
        if(start >= nums.length) return 0;

        return Math.max(helper(nums, start+1), nums[start] + helper(nums, start+2));
    }
    public int rob(int[] nums) {
        return helper(nums, 0);
    }
}

File: code/ch10/10.2.1-house-rob.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;

class Solution {
public:
    int helper(vector<int>& nums, int start) {
        if(start >= nums.size()) return 0;

        return max(helper(nums, start+1), nums[start] + helper(nums, start+2));
    }
    int rob(vector<int>& nums) {
        return helper(nums, 0);
    }
};